ads/openx.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, writeScript, checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function openx(global, data) {
+  const openxData = ['host', 'nc', 'auid', 'dfpSlot', 'dfp'];
+  const dfpData = Object.assign({}, data);// make a copy for dfp
+  checkData(data, openxData);
+
+  // consolidate doubleclick inputs for forwarding
+  // conversion rules are explained in openx.md
+  if (data.dfpSlot) {
+    // anything starting with 'dfp' gets promoted
+    for (const openxKey of openxData) {
```

Comment:
Unfortunately, we don't yet support `for/of` construct since it generates fairly inefficient es5 code. Could you please rephrase via `forEach` or `for/let i`